### PR TITLE
Fix regression in handling badly formed JSON

### PR DESCRIPTION
**Pull request by mcollina**
Fixes https://github.com/mercurius-js/mercurius/issues/677

### CODE REVIEW

**Comment by mcollina** [comment]
cc @simenb

**Review by simoneb — APPROVED** [review]

**Comment by SimenB** [comment]
Thanks for the quick turnaround! 👍

**Comment by mcollina** [comment]
No worries, this was quite serious. Thanks for spotting.

File: lib/routes.js [hunk 1]
```diff
@@ -135,14 +135,27 @@ module.exports = async function (app, opts) {
   const errorFormatter = typeof opts.errorFormatter === 'function' ? opts.errorFormatter : defaultErrorFormatter
 
   if (typeof opts.errorHandler === 'function') {
-    app.setErrorHandler(opts.errorHandler)
+    app.setErrorHandler((error, request, reply) => {
+      const errorHandler = opts.errorHandler
+      if (!request[kRequestContext]) {
+        // Generate the context for this request
+        request[kRequestContext] = { reply, app }
```
Comment (wiktor-obrebski) [inline lib/routes.js:142]: sorry for missing this.

@mcollina 
should not we try to generate full context, like in other places?

```ts
    if (contextFn) {
      request[kRequestContext] = await contextFn(request, reply)
      Object.assign(request[kRequestContext], { reply, app })
    } else {
      request[kRequestContext] = { reply, app }
    }
```

Comment (mcollina) [reply lib/routes.js:142]: I just focused on fixing it asap. Open a fresh PR in case.

**Pull request merged**